demos/classification_benchmark_demo/cpp_gapi/include/custom_kernels.hpp

```diff
@@ -0,0 +1,59 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <list>
+#include <map>
+#include <string>
+#include <vector>
+
+#include <cpp/ie_cnn_network.h>
+#include <ie_allocator.hpp>
+#include <ie_common.h>
+#include <ie_input_info.hpp>
+#include <opencv2/core.hpp>
+#include <opencv2/gapi/gkernel.hpp>
+#include <opencv2/gapi/gmat.hpp>
+
+
+struct IndexScore {
+    using IndicesStorage = std::list<size_t>;
+    using ConfidenceMap = std::multimap<float, typename IndicesStorage::iterator>;
+
+    using ClassDescription = std::tuple<float, size_t, std::string>;
+    using LabelsStorage = std::vector<ClassDescription>;
+
+    IndexScore() = default;
+
+    void getScoredLabels(const std::vector<std::string> &in_labes,
+                         LabelsStorage &out_scored_labels_to_append) const;
+    static IndexScore create_from_array(const float *out_blob_data_ptr, size_t out_blob_element_count,
+                                        size_t top_k_amount);
+private:
+    ConfidenceMap max_confidence_with_indices;
+    IndicesStorage max_element_indices;
+    size_t classification_indices_scale;
+};
+
+using GRect = cv::GOpaque<cv::Rect>;
+using GSize = cv::GOpaque<cv::Size>;
+using GIndexScore = cv::GOpaque<IndexScore>;
+
+namespace custom {
+G_API_OP(LocateROI, <GRect(GSize)>, "classification_benchmark.custom.locate-roi") {
```

Comment:
Why do we need to locate roi? I thought the inference is supposed to work on the full frame